src/main/java/de/dennisguse/opentracks/util/StringUtils.java

```diff
@@ -91,6 +96,77 @@ public static String formatDateTimeIso8601(long time_ms) {
         }
     }
 
+    /**
+     * Returns a datetime string in a readable user format.
+     *
+     * @param context       the context object.
+     * @param timeInMillis  datetime in milliseconds.
+     * @return              The formatted datetime string.
+     */
+    private static String getReadableDateTimeString(Context context, long timeInMillis) {
+        int flags = DateUtils.FORMAT_SHOW_DATE
+                | DateUtils.FORMAT_SHOW_TIME
+                | DateUtils.FORMAT_SHOW_YEAR
+                | DateUtils.FORMAT_SHOW_WEEKDAY;
+
+        return DateUtils.formatDateTime(context, timeInMillis, flags);
+    }
+
+    /**
+     * Returns the number of days since the epoch (January 01, 1970, 12:00 Midnight UTC).
+     *
+     * @param date date in milliseconds.
+     * @return the number of days in UTC time from the epoch.
+     */
+    public static long getDayNumber(long date) {
```

Comment:
Why do this computation manually and not using Java's default?
Date expiry = new Date(Long.parseLong(date));